src/Service.Tests/SqlTests/RestApiTests/Patch/PatchApiTestBase.cs

```diff
@@ -37,17 +37,16 @@ public virtual async Task PatchOne_Nulled_Test()
                 ""piecesAvailable"": null,
                 ""piecesRequired"": 4
             }";
-            string expectedLocationHeader = $"categoryid/3/pieceid/1";
 
             await SetupAndRunRestApiTest(
-                    primaryKeyRoute: expectedLocationHeader,
+                    primaryKeyRoute: $"categoryid/3/pieceid/1",
                     queryString: null,
                     entityNameOrPath: _Composite_NonAutoGenPK_EntityPath,
                     sqlQuery: GetQuery("PatchOne_Insert_Nulled_Test"),
                     operationType: EntityActionOperation.UpsertIncremental,
                     requestBody: requestBody,
                     expectedStatusCode: HttpStatusCode.Created,
-                    expectedLocationHeader: expectedLocationHeader
+                    expectedLocationHeader: string.Empty
```

Comment:
The primaryKeyRoute calculated would be `categoryid/3/pieceid/1`. However, this part is already present in the URL for Patch and Put requests. So, when constructing the complete URL, it is necessary to set this to `string.Empty`. 

Sample request:
![image](https://github.com/Azure/data-api-builder/assets/11196553/7ac726db-158d-43e2-a0e1-2cfcf9d817ac)

The tests are updated to validate that the primary key route is `string.Empty`

Reply:
Yes, in the current main/latest 0.8.51, the primary key route returned is incorrect. 

For a PATCH request, such as `https://localhost:5001/api/commodities/categoryid/101/pieceid/101`, when this results in an insert operation, the location header returned is `https://localhost:5001/api/commodities/categoryid/101/pieceid/101/categoryid/101/pieceid/101`
 
![image](https://github.com/Azure/data-api-builder/assets/11196553/5ed61056-1673-4cc3-87cb-8c53526c7a91)

This is because for PATCH and PUT operations already have the primary keys in the request URL. Calculating the primary keys again and appending them to the location header causes this inaccuracy. 

For PATCH and PUT requests, nothing needs to be appended. But, for POST requests, primary keys need to be appended because the request URL would not contain the primary keys. 

So, the tests for PATCH and PUT APIs will expect an empty string as the primary key route whereas tests for POST requests will expect that value to the primary key route

Reply:
Would this be a breaking change? Considering that this bug has not been reported yet, I think the users might not be relying on the location header, atleast for PUT/PATCH inserts, so we might be fine to introduce this. 
Since we were targeting to introduce all the breaking changes in 0.9-rc itself, it might just be the best time to make this change.

Reply:
I would consider this a bugfix and not a breaking change as the `Location` header was always returned. Yes, I think it should be fine to ship this fix in `0.9`  version